js/libs/form.js

```diff
@@ -97,10 +104,13 @@ function getDataSourceColumnValues(field) {
         return field.options;
       });
     });
-  })
-    .then(function(result) {
-      field.options = result;
-    });
+  }).then(function(result) {
+    field.options = result;
```

Comment:
You've used `getDataSourceColumnValues(field).then((result) => { field.options = result; })` many times in other places in this file, but this line is already here. One of them seems unnecessary (I think this one is).